docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,474 @@
+   
+.. image:: ./images/counteroffer-causality-ledgeraware-projection.svg
+   :align: center
+   :width: 100%
+
+The following points are worth highlighting:
+
+* In Alice's projection on the green ledger, Alice witnesses the archival of her `Iou`.
+  As explained in the :ref:`interop-ordering-guarantees` below,
+  the **Exercise** action is marked as merely being witnessed
+  in the transaction stream of a Participant Node that hosts Alice on the green ledger but not on the yellow ledger.
+  Similarly, the Painter merely witnesses the **Create** of his `Iou` in the Painter's projection on the green ledger.
+
+* In the Painter's projections, the `ShouIou` transaction `tx3` is unordered w.r.t. to the `CounterOffer` acceptance in `tx4`
+  like in the :ref:`case of ordinary causality graphs <counteroffer-causality-projections>`.
+  The edge `tx3` -> `tx4` is removed by the reduction step during projection.
+
+The projection of transfer actions can be illustrated with the :ref:`interoperable-causality-graph-linear`.
+The `A`-projections on the yellow and green ledger look as follows.
+The white color indicates that a transfer action has no incoming or outgoing ledger annotation.
+That is, a **Leave** action is white on the right hand side and an **Enter** action is white on the left hand side.
+
+.. https://app.lucidchart.com/documents/edit/edbf9aaf-b7da-4e68-b9c9-9e631c3a87bb
+
+.. image:: ./images/transfer-projection.svg
+   :align: center
+   :width: 100%
+
+.. _interop-ordering-guarantees:
+
+Ledger API ordering guarantees
+******************************
+
+The Transaction Service and the Active Contract Service are derived from the local ledger that the Participant Node maintains for the party.
+Let `Y` be the set of ledgers on which the Participant Node hosts a party.
+The transaction tree stream outputs a topological sort of the party's local ledger on `Y`, with the following modifications:
+
+#. **Transfer** actions with either an incoming or an outgoing ledger annotation are output as **Enter** and **Leave** events.
+   **Transfer** actions with both incoming and outgoing ledger annotations are omitted.
+
+#. The incoming and outgoing ledger annotations are not output.
+   Transaction actions with an incoming or outgoing ledger annotation
+   that is not in `Y` are marked as merely being witnessed if the
+   party is an informee of the action.
+
+#. **Fetch** nodes and **NoSuchKey** are omitted.
+
+The flat transaction stream contains precisely the ``CreatedEvent``\ s, ``ArchivedEvent``\ s, and the **Enter** and **Leave** actions that correspond to **Create**, consuming **Exercise**, **Enter** and **Leave** actions in transaction trees on the transaction tree stream where the party is a stakeholder of the affected contract and that are not marked as merely being witnessed.
+
+Similarly, the active contract service provides the set of contracts that are active at the returned offset according to the Transaction Service streams.
```

Comment:
Maybe clarify that the "Transaction Service" is the same as the "flat transaction stream".

Reply:
It's not the same (you can also read off activeness from the tree stream), but it doesn't matter here. Changed to flat transaction stream.